src/tif_model_iterator/tif_kernel_iterator.py

```diff
@@ -98,12 +119,20 @@ def predict_all_output(
         self._write_full_gdf_to_file()
         self._clean_up_part_files()
 
+    def __check_if_part_exists(self, afilepath):
+        """
+        If a rain has failed some part do not have to be redone, this functions check if they exist
```

Comment:
What do you mean by "a rain"?

Reply:
part instead of rain, maybe i need a vacation to the sun